src/coreclr/jit/emitxarch.cpp

```diff
@@ -2569,6 +2321,152 @@ inline bool hasCodeMR(instruction ins)
     return ((insCodesMR[ins] != BAD_CODE));
 }
 
+//------------------------------------------------------------------------
+// emitGetVexPrefixSize: Gets Size of VEX prefix in bytes
+//
+// Arguments:
+//    id   -- The instruction descriptor
+//
+// Returns:
+//    Prefix size in bytes.
+//
+unsigned emitter::emitGetVexPrefixSize(instrDesc* id)
+{
+    instruction ins  = id->idIns();
+    emitAttr    size = id->idOpSize();
+
+    assert(IsVexEncodedInstruction(ins));
+
+    if (EncodedBySSE38orSSE3A(ins))
+    {
+        // When the prefix is 0x0F38 or 0x0F3A, we must use the 3-byte encoding
```

Comment:
This filters out a majority of the complex instructions, particularly those that take 3 inputs or do other "special things" with register representation.